File: paper_artifacts/experiments/llr8/artifacts/sources/oss120b-c/tsvc_2_vag__submitted.c
#include <stddef.h>
#include <stdint.h>

void tsvc_2_vag_fp64(double *restrict a, const double *restrict b,
                      const int32_t *restrict ip, int64_t LEN_1D,
                      uint8_t *restrict workspace, int64_t workspace_bytes)
{
    (void)workspace;
    (void)workspace_bytes;
    #pragma omp parallel for simd schedule(static)
    for (int64_t i = 0; i < LEN_1D; ++i) {
        a[i] = b[ip[i]];
    }
}

File: paper_artifacts/experiments/llr8/artifacts/sources/oss120b-fortran-skills/tsvc_2_vag__submitted.f90
subroutine tsvc_2_vag_fp64(a, b, ip, len_1d, workspace, workspace_size) bind(C)
  use iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: len_1d, workspace_size
  type(c_ptr), intent(inout) :: workspace
  real(c_double), intent(inout) :: a(len_1d)
  real(c_double), intent(in) :: b(len_1d)
  integer(c_int32_t), intent(in) :: ip(len_1d)
  integer(c_int64_t) :: i
  !$omp parallel do simd schedule(static)
  do i = 1, len_1d
    a(i) = b(ip(i))
  end do
end subroutine tsvc_2_vag_fp64